utils/yoke/src/kinda_sorta_dangling.rs

```diff
@@ -0,0 +1,56 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::mem::MaybeUninit;
+use core::ops::{Deref, DerefMut};
+
+/// This type is intended to be similar to the type `MaybeDangling<T>`
+/// proposed in [RFC 3336].
+///
+/// The effect of this is that in Rust's safety model, types inside here are not
+/// expected to have any memory dependent validity properties (`dereferenceable`, `noalias`).
+///
+/// See [#3696] for a testcase where `Yoke` fails this under miri's field-retagging mode.
+///
+/// This has `T: 'static` since we don't need anything
+/// else and we don't want to have to think (more) about variance.
+///
+/// After [RFC 3336] lands we can use `MaybeDangling` instead.
+///
+/// [RFC 3336]: https://github.com/rust-lang/rfcs/pull/3336
+/// [#3696]: https://github.com/unicode-org/icu4x/issues/3696
+#[repr(transparent)]
+pub(crate) struct KindaSortaDangling<T: 'static> {
```

Comment:
Issue: Do you need to implement `Drop`?

Reply:
Please add a test, if we don't already have one, with `Yoke<Cow<'static, str>, &str>` where you put an owned string in the Yoke and make sure it doesn't leak.

Reply:
oh. yes, I was going to do that and forgot